### DISCUSSION TITLE
Can I trigger an exception on missing fields?

### DISCUSSION THREAD

**Discussion by stdNullPtr** [aaubry/YamlDotNet#813] [Q&A] [answered]
Hey, so I currently have a yaml with 3 fields, doesn't matter what their names or values.
And let's say I have an object to deserialize to that has 4 fields. I would expect that if the 4th field is not present in the yaml, I would get an exception - but that's not the case.
Any ideas?

**Comment by stdNullPtr** [comment]
Also, can I generate a comment for every field, during serialization? For example, adding a comment in the field with YamlMember or something?

**Comment by EdwardCooke** [comment]
You wouldn’t get an exception when your yaml has 3 fields and your class has 4. You would only get an exception a field in your yaml doesn’t exist on your class. 

You should be able to use the yamlmember attribute on fields and properties to add comments in your yaml without issue.

**Reply by stdNullPtr** [reply]
So you would have to check after parsing if the fields are populated? I thought there would be a more elegant approach.

Can you please point me to the relevant documentation about the yamlMember attributes? Thanks!

**Comment by EdwardCooke** [comment, marked as answer]
There’s pretty good comments on the yamlmemberattribute that you can review. They should also show up in intelligence in visual studio. 

https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Serialization/YamlMemberAttribute.cs

I don’t think there’s any formal documentation on it.

**Reply by stdNullPtr** [reply]
You are right the comments are good, thank you

**Comment by EdwardCooke** [comment]
You would also need to check if the fields get set by yourself. One potential reliable solution would be to use properties with a setter that keep track when the property was set.

**Reply by stdNullPtr** [reply]
Thanks that sounds like an idea
